File: sdks/node/serto/test/fixtures.js
'use strict';

// Handlers referenced by harness.test.js via file-path entrypoints. (No tests here; harmless if the
// runner loads it.)

module.exports.success = async (ctx) => {
  ctx.logger.info('ran ok');
  ctx.publish('test.subject', { k: 1 });
};

module.exports.failing = async () => {
  throw new Error('boom');
};

module.exports.secret = async (ctx) => {
  ctx.logger.info('secret=' + (ctx.secrets.API_KEY || ''));
};

File: sdks/python/tests/fixtures.py
"""Handlers referenced by test_harness via file-path entrypoints."""


def success_handler(ctx):
    ctx.logger.info("ran ok")
    ctx.publish("test.subject", {"k": 1})


def secret_handler(ctx):
    ctx.logger.info("secret=" + ctx.secrets.get("API_KEY", ""))


def failing_handler(ctx):
    raise RuntimeError("boom")


async def async_handler(ctx):
    ctx.logger.info("async ran")
